common/reflect_resource_plugin_framework.go

```diff
@@ -503,3 +610,14 @@ func PluginFrameworkResourceStructToSchemaMap(v any, customizeSchema func(Custom
 		return attributes
 	}
 }
+
+func PluginFrameworkDataSourceStructToSchemaMap(v any, customizeSchema func(CustomizableSchemaPluginFrameworkDataSource) CustomizableSchemaPluginFrameworkDataSource) map[string]dataschema.Attribute {
+	attributes := pluginFrameworkDataSourceTypeToSchema(reflect.ValueOf(v))
+
+	if customizeSchema != nil {
+		cs := customizeSchema(*ConstructCustomizableSchemaDataSource(attributes))
+		return cs.ToAttributeMap()
+	} else {
+		return attributes
+	}
+}
```

Comment:
I am not sure if duplicating everything is a good idea. But the other option is to create if else branches in a single function, might also be pretty ugly.